HACKING.md

```diff
@@ -14,3 +16,106 @@ echo "Ending pre-commit"
 ````
 
 into the `.git/hooks/pre-commit` file.
+
+#### Makefile
+
+* After making a change in the code run `$ make agda_apia_changed`
+* Clean and install Apia. `$ make install-bin`
+* Run Hlint. `$ make hlint`
+* Check the white spaces. `$ make check-whitespace`
+* Generate all. `$ make generated_all`
+* Errors. `$ make errors`
+* Documentation. `$ make haddock`
+* Clean. `$make clean`
+
+
+#### Code Style
+
+* Run always `$ make hlint`
+* Delete *trailing spaces*
+* Use *spaces* instead of tabs
+* Use *two spaces* per each indentation.
+* Each line can not exceed *80 characters.*
+* A file must end by a *newline*.
+* The imports must be *sorted* alphabetically.
+* Try to avoid the *parenthesis* when it is possible.
+  * Use the dollar symbol **$** to avoid redundant parenthesis.
+    Instead of
+
+    ```Haskell
+    let result = f ( g (x `div` y) )
+    ```
+
+    Why not?
+
+    ```Haskell
+    let result = f $ g $ x `div` y
+    ```
```

Comment:
No. Actually, I think it could be `let result = f . g $ x `div` y` better, don't you?